lib/cocoapods/installer/analyzer.rb

```diff
@@ -554,11 +563,20 @@ def filter_pod_targets_for_target_definition(target_definition, pod_targets_by_t
             whitelists = dependencies.map do |dependency|
               target_definition.pod_whitelisted_for_configuration?(dependency.name, configuration_name)
             end.uniq
+            if resolver_spec.transitive?
+              # Include the whitelist config for the source of the transitive dependency, so we can exclude
+              # the transitive dependencies of whitelisted dependencies
+              whitelists += resolver_spec.dependent_specs.map do |dependent_spec|
+                target_definition.pod_whitelisted_for_configuration?(dependent_spec.name, configuration_name)
+              end
+              whitelists.uniq!
+            end
 
             case whitelists
             when [], [true] then nil
             when [false] then next
             else
+              next if resolver_spec.transitive?
```

Comment:
this is incorrect, we need to allow `[false, true]` iff the spec is `transitive?` rather than skipping with `next`. this would resolve the above issue for when one dependent is whitelisted and the other is not